### DISCUSSION TITLE
Create Repo returning 403?

### DISCUSSION THREAD

**Discussion by FlorianEagox** [community/community#24348] [Apps, API and Webhooks]
<p>I’ve also asked this question on StackOverflow here:<br>
</p><aside class="onebox stackexchange">
  <header class="source">
      <a href="https://stackoverflow.com/questions/64563009/github-api-create-repo-returning-403" target="_blank" rel="noopener nofollow ugc">stackoverflow.com</a>
  </header>
  <article class="onebox-body">
      <a href="https://stackoverflow.com/users/5038503/seth-painter" target="_blank" rel="noopener nofollow ugc">
    <img alt="Seth Painter" src="https://user-images.githubusercontent.com/5177040/181084316-d45c98bd-6672-4fad-87fa-4e7ebda3a0e6.png" class="thumbnail onebox-avatar" width="128" height="128">
  </a>
<h4>
  <a href="https://stackoverflow.com/questions/64563009/github-api-create-repo-returning-403" target="_blank" rel="noopener nofollow ugc">GitHub API Create Repo returning 403?</a>
</h4>

<div class="tags">
  <strong>github, request, github-api, github-api-v3</strong>
</div>

<div class="date">
  asked by
  
  <a href="https://stackoverflow.com/users/5038503/seth-painter" target="_blank" rel="noopener nofollow ugc">
    Seth Painter
  </a>
  on <a href="https://stackoverflow.com/questions/64563009/github-api-create-repo-returning-403" target="_blank" rel="noopener nofollow ugc">09:15PM - 27 Oct 20 UTC</a>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I’m making a GitHub app, and I’m trying to create a repo according to this documentation</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-a-repository-for-the-authenticated-user" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-a-repository-for-the-authenticated-user" target="_blank" rel="noopener nofollow ugc">Repositories - GitHub Docs</a></h3>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>When I make a GET request to that URL, it works and I get the user’s repos, but making a post request like this:</p>
<pre><code class="lang-auto">curl --request POST \
  --url https://api.github.com/user/repos \
  --header 'authorization: token  [user_access_token]' \
  --data '{"name": "test"}'
</code></pre>
<p>Returns this</p>
<pre><code class="lang-auto">{
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/reference/repos#create-a-repository-for-the-authenticated-user"
}
</code></pre>
<p>My app has these permissions<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/d/7d530b9a8181baf3bfcdb16bfc155b84dd50b129.png" data-download-href="/uploads/short-url/hSFBFBqpFMicxV3GwzdDc5KiXBn.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/5177040/181084317-934bc596-0ad8-457d-a9fd-85547d23c4e3.png" alt="image" data-base62-sha1="hSFBFBqpFMicxV3GwzdDc5KiXBn" width="690" height="357" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/d/7d530b9a8181baf3bfcdb16bfc155b84dd50b129_2_690x357.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/d/7d530b9a8181baf3bfcdb16bfc155b84dd50b129_2_1035x535.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/7/d/7d530b9a8181baf3bfcdb16bfc155b84dd50b129.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/d/7d530b9a8181baf3bfcdb16bfc155b84dd50b129_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1087×563 47.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by nethgato** [comment, marked as answer]
<p>Hey there <a class="mention" href="/thefoxarmy">@TheFoxarmy</a> :wave:</p>
<p>Thanks so much for joining our Community and for submitting your post. I appreciate the details you’ve provided, though I think there may be a confusion about where that <code>scope</code>  needs to be defined.</p>
<p>You’ve shown us the permissions that the repository itself has in relation to the installation of your app, but the <code>repo</code> <code>scope</code> needs to be defined for the application’s token and not within the repository’s permissions.</p>
<p>Relevant documentation for this lives here:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps#available-scopes" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps#available-scopes" target="_blank" rel="noopener">Scopes for OAuth Apps - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>This should be what you need!</p>
<p>If you can verify that the token that your app leverages has the appropriate scope defined and you are <em>still</em> experiencing this problem, please let us know. :bow:</p>

**Comment by FlorianEagox** [comment]
<p>Hi, so your post and further reading about this subject has actually lead me to conclude that creating repos is a feature of OAuth Apps, not GitHub Apps?<br>
I believe this is the case, and after switching to using an OAuth app instead of a GitHub app, I am no longer experiencing this problem and haven’t found any loss in functionality with my app.<br>
If this is the case, I would hope that going forward, the documentation becomes more clear about what can and should be done with each authentication strategy for each API function.</p>
